core/test/network-records-to-devtools-log.js

```diff
@@ -225,28 +242,87 @@ function addRedirectResponseIfNeeded(networkRecords, record) {
  * @return {LH.DevtoolsLog}
  */
 function networkRecordsToDevtoolsLog(networkRecords, options = {}) {
+  // Clone test network records objects before potential modifications.
+  networkRecords = networkRecords.map(record => {
+    if (record.constructor === Object) {
```

Comment:
I can remove this object literal check and the one below and tests still all pass. I believe only the `'should roundtrip a real devtools log properly'` test actual passes in real NetworkRequest instances. It seems wrong to modify those though, hence the checks here.